source/deployment/sso-gitlab.rst

```diff
@@ -0,0 +1,56 @@
+GitLab Single Sign-On
+=====================
+
+Configure Mattermost to use GitLab as a single sign-on (SSO) service for team creation, account creation, and user sign-in.
+
+Step 1: Add an OAuth application to your GitLab account
+-------------------------------------------------------
+
+1. Sign in to your GitLab account and go to ``https://{gitlab-site-name}/profile/applications``. For *{gitlab-site-name}* use the name of your GitLab instance. If you are using GitLab itself as your OAuth provider, use *gitlab.com*.
+2. Add a new application:
+
+  a. In the **Name** field, type *Mattermost*.
+  b. In the **Redirect URI** field add the following two lines, using your own value for *{mattermost-site-name}*.
+
+    .. code-block:: text
+
+      https://{mattermost-site-name}/login/gitlab/complete
+      https://{mattermost-site-name}/signup/gitlab/complete
+
+  c. In the *Scopes* section, select **api**.
+
+3. Click **Save application**.
+
+Keep the GitLab window open because you need the *Application Id* and *Secret* when you configure Mattermost.
+
+Step 2: Configure Mattermost for GitLab SSO
+-------------------------------------------
+
+1. On your Mattermost server, add the *Application Id* and the *Secret* to the *GitLab* settings section in the ``config.json`` file:
+
+  a. Open ``config.json`` as root in a text editor. It's usually in ``/opt/mattermost/config`` but might be elsewhere on your system.
+  b. Locate the *GitLabSettings* section and add or update the following information:
+
+    .. code-block:: javascript
+
+      "GitLabSettings": {
+          "Enable": true,
+          "Secret": "{mattermost-app-secret-from-gitlab}",
+          "Id": "{mattermost-app-application-id-from-gitlab}",
+          "Scope": "",
+          "AuthEndpoint": "https://{gitlab-site-name}/oauth/authorize",
```

Comment:
SAme as above, is it always https? see this comment: https://github.com/mattermost/platform/issues/6489#issuecomment-304810961